azurerm/internal/services/containers/container_registry_resource.go

```diff
@@ -804,6 +851,18 @@ func flattenNetworkRuleSet(networkRuleSet *containerregistry.NetworkRuleSet) []i
 	return []interface{}{values}
 }
 
+func flattenQuarantinePolicy(p *containerregistry.Policies) []interface{} {
+	if p == nil || p.QuarantinePolicy == nil {
+		return nil
+	}
+
+	t := *p.QuarantinePolicy
+	quarantinePolicy := make(map[string]interface{})
+	enabled := strings.EqualFold(string(t.Status), string(containerregistry.Enabled))
```

Comment:
this shouldn't be doing a case insensitive comparison - that would hide an API bug which needs to be fixed, so we can reduce this to:

```suggestion
	enabled := t.Status == containerregistry.Enabled
```